File: Model/Eintites/Movie.cs
namespace Movie_Api.Model.Eintites
{
    public class Movie
    {
        public int Id { get; set; }
        public string? Title { get; set; }
        public int Year { get; set; }
        public double Rate { get; set; }
        public string? Storyline { get; set; }
        public byte[]? Poster { get; set; }

        public int GenraId { get; set; }

        public Genra? Genra { get; set; }


    }
}

File: movie-frontend/src/types/movie.ts
export interface Genre {
  id: number;
  name: string;
}

export interface Movie {
  id: number;
  title: string;
  year: number;
  rate: number;
  storyline: string;
  poster?: number[] | null;
  posterUrl?: string | null;
  genraId: number;
  genraName?: string;
}

export interface CreateMovieDto {
  title: string;
  year: number;
  rate: number;
  storyline: string;
  poster: File;
  genraId: number;
}

export interface UpdateMovieDto {
  title?: string;
  year?: number;
  rate?: number;
  storyline?: string;
  poster?: File;
  genraId?: number;
}

export interface ApiResponse<T> {
  success: boolean;
  massage: string;
  version?: string;
  data: T;
  error?: any;
}